include/boost/python/detail/type_traits.hpp

```diff
@@ -0,0 +1,105 @@
+// Copyright David Abrahams 2002.
+// Distributed under the Boost Software License, Version 1.0. (See
+// accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_DETAIL_TYPE_TRAITS_HPP
+# define BOOST_DETAIL_TYPE_TRAITS_HPP
```

Comment:
This should be BOOST_PYTHON_DETAIL_TYPE_TRAITS_HPP, to really match the complete filename.